projects/packages/forms/src/contact-form/class-contact-form-endpoint.php

```diff
@@ -630,46 +665,40 @@ public function get_all_integrations_status( $request ) {
 	 * @return array Service status data.
 	 */
 	private function get_service_status( $slug ) {
+		$config = $this->get_supported_integrations()[ $slug ];
+
+		// Default response for all integrations
+		$response = array(
+			'type'        => $config['type'],
+			'slug'        => $slug,
+			'isConnected' => false,
+			'settingsUrl' => isset( $config['settings_url'] ) ? $config['settings_url'] : null,
+			'pluginFile'  => null,
+			'isInstalled' => false,
+			'isActive'    => false,
+			'version'     => null,
+			'details'     => array(),
+		);
+
+		// Process settingsUrl to return a full URL if present (for services only)
+		if ( $response['settingsUrl'] ) {
+			$response['settingsUrl'] = esc_url( Redirect::get_url( $response['settingsUrl'] ) );
+		}
+
 		switch ( $slug ) {
+			case 'google-drive':
+				$user_id                 = get_current_user_id();
+				$jetpack_connected       = ( new Host() )->is_wpcom_simple() || ( new Connection_Manager( 'jetpack-forms' ) )->is_user_connected( $user_id );
+				$is_connected            = $jetpack_connected && Google_Drive::has_valid_connection( $user_id );
+				$response['isConnected'] = $is_connected;
+				break;
 			case 'salesforce':
-				return array(
-					'type'        => 'service',
-					'slug'        => 'salesforce',
-					'pluginFile'  => null,
-					'isInstalled' => false,
-					'isActive'    => false,
-					'isConnected' => false,
-					'version'     => null,
-					'settingsUrl' => null,
-					'details'     => array(),
-				);
-			default:
-				return array(
-					'type'        => 'service',
-					'slug'        => $slug,
-					'pluginFile'  => null,
-					'isInstalled' => false,
-					'isActive'    => false,
-					'isConnected' => false,
-					'version'     => null,
-					'settingsUrl' => null,
-					'details'     => array(),
-				);
+				// No overrides needed for now; keep defaults.
+				break;
+			// Add other service cases as needed.
```

Comment:
I updated this method to defined some defaults first, and then update the defaults only if needed for each integration. It's more DRY.